modules/andes-core/broker/src/main/java/org/wso2/andes/kernel/OnflightMessageTracker.java

```diff
@@ -365,6 +383,11 @@ public boolean handleAckReceived(UUID channel, long messageID) throws AndesExcep
             log.debug("Ack Received message id= " + messageID + " channel id= " + channel);
         }
 
+        if (retainedMessageList.contains(messageID)) {
```

Comment:
We could only use the remove here. and check whether it returns a value or null and validate